test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -48,9 +40,7 @@ public void IrisLightGbm()
 
             var loader = mlContext.Data.CreateDatabaseLoader(loaderColumns);
 
-            var databaseSource = new DatabaseSource(SqlClientFactory.Instance, connectionString, commandText);
-
-            var trainingData = loader.Load(databaseSource);
+            var trainingData = loader.Load(GetIrisDatabaseSource("SELECT * FROM "));
 
```

Comment:
suggestion: The "SELECT * FROM " is an incomplete sql statement and looks a bit odd. Would it be better to pass in a string format string and then call string.format in the GetIrisDatabaseSource function?

Reply:
Can you please clarify what you mean by a "string format string"? I agree that having an incomplete SQL statement looks a bit odd, but I think this is a good way to keep code duplication at a minimum.

One way that can also work is:
```csharp
var trainingData = loader.Load(GetIrisDatabaseSource("SELECT * FROM _database_"));
```
...
```csharp
private DatabaseSource GetIrisDatabaseSource(string command)
{
    if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
        return new DatabaseSource(
            SqlClientFactory.Instance,
            GetMSSQLConnectionString(TestDatasets.irisDb.name),
            command.Replace("_database_", TestDatasets.irisDb.trainFilename));
    else
        return new DatabaseSource(
            SQLiteFactory.Instance,
            GetSQLiteConnectionString(TestDatasets.irisDbSQLite.name),
            command.Replace("_database_", TestDatasets.irisDbSQLite.trainFilename));
}
```

Reply:
Please take a look at [https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netcore-3.1)

---
In reply to: [419013429](https://github.com/dotnet/machinelearning/pull/5080#discussion_r419013429) [](ancestors = 419013429)